### ISSUE TITLE
[Feature] Playgrounds

### ISSUE THREAD

**Issue by skoropadas** [skoropadas/ng-doc#12]
### Description

Before the NgDoc release, the playground function was removed, because it forced to turn off the `aot` compilation mode and its API did not quite suit me.

This feature can be extremely useful, especially for the presentation of components and directives, based on the passed class and template, it automatically does the following:

- it automatically determines the list and types of inputs and creates controls for them that you can use to manipulate the value of the input
- generates view and its template based on values of controls
- supports the creation of custom controls for data types that NgDoc cannot detect automatically

### Proposed solution

It should be implemented as an additional property in `ng-doc.dependencies.ts` config file.

```typescript
import {NgDocDependencies} from '@ng-doc/builder';
import {NgDocTagComponent} from '@ng-doc/ui-kit';

import {PlaygroundPageModule} from './ng-doc.module';

const PlaygroundPageDependencies: NgDocDependencies = {
	// Add your module here
	module: PlaygroundPageModule,
	playgrounds: {
		/**
		 * Playground key that should be used in the template to render the playground
		 */
		TagIconPlayground: {
			/**
			 * The component or directive class that should be rendered
			 */
			target: NgDocTagComponent,
			/**
			 * Template that should be used to render the component playground.
			 * `ng-doc-selector` is dynamic tag that will be replaced with the component selector,
			 * it's required to render the components or directive that have multiple selectors.
			 */
			template: `
			<ng-doc-selector>
				{{iconContent}}
				{{data.label}}
			</ng-doc-selector>`,
			/**
			 * Dynamic content that can be used in the template.
			 * It will be represented as a toggle button in the playground form and added to the
			 * template when it's toggled.
			 */
			dynamicContent: {
				iconContent: {
					label: 'icon',
					template: '<ng-doc-icon icon="info" [size]="16"></ng-doc-icon>'
				}
			},
			/**
			 * Custom data that can be used in the template to fill components with data.
			 */
			data: {
				label: 'Tag Label',
			}
		},
	}
}

export default PlaygroundPageDependencies;
```

Can be rendered on the page via `NgDocActions` like that:

```markdown
## Playground

{{ NgDocActions.playground("TagIconPlayground") }}

```

So here is an example of how it should work:

https://user-images.githubusercontent.com/20700969/218282909-ada4047a-ed41-41ce-8d2c-4f39904dc410.mov

### Alternatives considered

nope.

**Comment by R-Bower** [comment]
Looking very forward to this feature!

**Comment by skoropadas** [comment]
Released in 15.3.1 (15.3.0 was stuck because of npm authorization)
